operator/pkg/values/translation/translation.go

```diff
@@ -123,40 +124,123 @@ func GetGlobalMonitoring(m *platform.GlobalMonitoring) *ValuesBuilder {
+	if scannerComponent != nil {
+		switch *scannerComponent {
+		case platform.ScannerComponentDisabled:
+			sv.SetBoolValue("disable", true)
+		case platform.ScannerComponentEnabled:
+			sv.SetBoolValue("disable", false)
+		default:
+			sv.SetError(fmt.Errorf("invalid ScannerComponentPolicy %q", *scannerComponent))
 		}
-		autoscaling.SetInt32("minReplicas", scaling.MinReplicas)
-		autoscaling.SetInt32("maxReplicas", scaling.MaxReplicas)
-		sv.AddChild("autoscaling", &autoscaling)
 	}
+}
 
-	if analyzer != nil {
-		sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
-		sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
-		sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
+// SetScannerAnalyzerValues sets values in "sv" based on "analyzer".
+func SetScannerAnalyzerValues(sv *ValuesBuilder, analyzer *platform.ScannerAnalyzerComponent) {
+	if analyzer == nil {
+		return
 	}
+	setScannerComponentScaling(sv, analyzer.GetScaling())
+	sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
+	sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
+	sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
 }
 
-// SetScannerDBValues sets values in "sb" based on "db".
+// SetScannerDBValues sets values in "sv" based on "db".
 func SetScannerDBValues(sv *ValuesBuilder, db *platform.DeploymentSpec) {
 	if db != nil {
 		sv.SetStringMap("dbNodeSelector", db.NodeSelector)
 		sv.AddChild("dbResources", GetResources(db.Resources))
 		sv.AddAllFrom(GetTolerations("dbTolerations", db.Tolerations))
 	}
 }
+
+// SetScannerV4DBValues sets values in "sv" based on "db"
+func SetScannerV4DBValues(sv *ValuesBuilder, db *platform.ScannerV4DB) {
+	if db == nil {
+		return
+	}
+
+	dbVB := NewValuesBuilder()
+	dbVB.SetStringMap("nodeSelector", db.NodeSelector)
+	dbVB.AddChild(ResourcesKey, GetResources(db.Resources))
+	dbVB.AddAllFrom(GetTolerations(TolerationsKey, db.Tolerations))
+	setScannerV4DBPersistence(&dbVB, db.Persistence)
+
+	sv.AddChild("db", &dbVB)
+}
+
+func setScannerV4DBPersistence(sv *ValuesBuilder, persistence *platform.Persistence) {
+	if persistence == nil {
+		return
+	}
+
+	hostPath := persistence.GetHostPath()
+	pvc := persistence.GetPersistentVolumeClaim()
+
+	if hostPath != "" && pvc != nil {
+		sv.SetError(errors.New("invalid persistence configuration, either hostPath or persistentVolumeClaim must be set, not both"))
+		return
+	}
+
+	persistenceVB := NewValuesBuilder()
+	if hostPath != "" {
+		persistenceVB.SetStringValue("hostPath", hostPath)
+	}
+
+	if pvc != nil {
+		// Unlike central-db's PVC we don't use the extension.ReconcilePVCExtension.
+		// The operator creates this PVC through the helm chart. The difference is
+		// that the extension prevents central DB's PVC deletion on deletion of the CR.
+		// Since scanner V4's DB contains data which recovers by itself
+		// in case of a new CR creation, it is safe to remove the PVC.
```

Comment:
safe, but not preferred. Does this imply it will always be cleared upon upgrade or something like that?